packages/core/src/common/message-service-protocol.ts

```diff
@@ -53,6 +87,25 @@ export class MessageClient {
         this.logger.info(message.text);
         return Promise.resolve(undefined);
     }
+
+    /**
+     * Show progress message with possible actions to user.
+     *
+     * To be implemented by an extension, e.g. by the messages extension.
+     */
+    showProgress(progressId: string, message: ProgressMessage, cancellationToken: CancellationToken): Promise<string | undefined> {
+        this.logger.info(message.text);
+        return Promise.resolve(undefined);
+    }
+
+    /**
+     * Update started progress message.
+     *
+     * To be implemented by an extension, e.g. by the messages extension.
+     */
+    reportProgress(progressId: string, update: ProgressUpdate): Promise<void> {
```

Comment:
@vinokurig, please add `message: ProgressMessage` as parameter, and adjust the tail.

`reportProgress(progressId: string, message: ProgressMessage, update: ProgressUpdate): Promise<void> `

I know, this looks a bit redundant, but I fear that's the only way how we can respawn the progress on FE reload if the source is on the BE (thinking loud about the git clone case.)

Reply:
done